making-mods/advanced-js/mutation-observers.md

```diff
@@ -0,0 +1,199 @@
+
+### Configuration options (`configurationOptions`)
+
+The configuration options are a JavaScript `object` that sets up the observer to watch for the appropriate types of mutations on the target element. There are many different options, but the 2 most important to mods are `attributes` and `childList`. A more complete explanation of all the options can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#parameters).
+
+- Setting `configurationOptions = {attributes: true}` will set the observer to watch for changes to the attributes of the `targetElement`.
+
+  - This is particularly useful for detecting changes in state represented by adding or removing a class on an element. Like with classes such as `.active` or `.hidden`.
+  - The `#browser` div in Vivaldi contains several state based classes that reflect settings the user has selected.
+
+- Setting `configurationOptions = {childList: true}` will set the observer to watch for child elements of the `targetElement` that are added or removed.
+
+  - Sometimes elements are not hidden or transformed into something else when a change to the UI is made. In these cases, new elements are added, so an observer of this type is useful to detect those changes.
+  - Adding `subtree: true` as well, like so `{childList: true, subtree: true}`, will extend the observer to all decedent elements of the `targetElement`rather than only the first layer of children.
+    - So all the elements below the `.parent` will be observed with this configuration (denoted by the class `observed`):
+
+      ```HTML
+      <div class="parent targetElement">
+        <div class="child observed">
+          <div class="grandChild observed"> ... </div>
+        </div>
+        <div class="child observed"> ... </div>
+      </div>
+
+      <div class="notObserved"> ... </div>
+      ```
+
+### Handler function for mutations (`callBack()`)
+
+The handler function is run whenever a mutation of the types you configured is detected.
+
+The list of mutations should be passed in as a parameter to allow the function to perform different actions depending on the contents of the mutations. In the example, this parameter is named `mutations` and included like so:
+
+```JavaScript
+function callBack (mutations) {
+  ...
+```
+
+There is often more than one mutation included in the list, so it is important to check each one individually. This is accomplished with a `forEach` loop in the example (usage information can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)).
+
+Once an individual mutation is isolated, you can check what type of mutation it is by looking at the `type` attribute of the mutation object ( `if (mutation.type === ... ) { ...` ). The types that are possible are determined by the configuration of the observer. The `type` could be equal to `attribute` or `childList` based on the configuration options outlined above.
```

Comment:
Do you need ‘equal to’? & maybe you could add quotes around the values.

Reply:
Like in e1bb94f?

Reply:
Like `"attribute"`, to make it clear it’s an exact string (emphasis on both ‘exact’ and ‘string’).